rslint_core/src/groups/errors/use_isnan.rs

```diff
@@ -0,0 +1,221 @@
+                    .primary(expr.range(), "")
+                    .note("note: `NaN` is not equal to anything including itself");
+
+                // telling the user to use isNaN for `<`, `>`, etc is a bit misleading so we won't do it if that is the case
+                if op == op!(==) || op == op!(===) {
+                    err = err.note(format!(
+                        "help: use `isNaN` instead: `{}`",
+                        color(&format!("isNaN({})", opposite))
+                    ))
+                } else if op == op!(!=) || op == op!(!==) {
+                    err = err.note(format!(
+                        "help: use `isNaN` instead: `{}`",
+                        color(&format!("!isNaN({})", opposite))
+                    ))
+                }
+
+                ctx.add_err(err);
+            }
+            SWITCH_STMT if self.enforce_for_switch_case => {
+                let stmt = node.to::<SwitchStmt>();
+                let expr = stmt.test()?.condition()?;
+                if expr.text() == "NaN" {
+                    let err = ctx
+                        .err(
+                            self.name(),
+                            "a switch statement with a test of `NaN` will never match",
+                        )
+                        .primary(expr.range(), "")
+                        .note("note: `NaN` is not equal to anything including itself");
+
+                    ctx.add_err(err);
+                }
+            }
+            CASE_CLAUSE if self.enforce_for_switch_case => {
+                let case = node.to::<CaseClause>();
+                let expr = case.test()?;
+                if expr.text() == "NaN" {
+                    let err = ctx
+                        .err(self.name(), "a case with a test of `NaN` will never match")
+                        .primary(expr.range(), "")
+                        .note("note: `NaN` is not equal to anything including itself");
+
+                    ctx.add_err(err);
+                }
+            }
+            CALL_EXPR if self.enforce_for_index_of => {
+                let expr = node.to::<CallExpr>();
+                let callee = expr.callee()?;
+                if is_indexof_static_prop(&callee)
+                    && expr.arguments()?.args().next()?.text() == "NaN"
+                {
+                    let err = ctx
+                        .err(
+                            "use-isnan",
+                            "an index check with `NaN` will always return `-1`",
+                        )
+                        .primary(expr.range(), "")
+                        .note("help: index checks use `===` internally, which will never match because `NaN` is not equal to anything");
+
+                    ctx.add_err(err);
+                }
+            }
+            _ => {}
+        }
+        None
+    }
+}
+
+const INDEX_OF_NAMES: [&str; 2] = ["lastIndexOf", "indexOf"];
+
+fn is_indexof_static_prop(expr: &Expr) -> bool {
+    match expr {
```

Comment:
You can also call the indexOf methods like this: `Array.prototype.indexOf`, which is not catched here I think.

Reply:
`Array.prototype.indexOf(NaN);` is caught, which im not sure is ideal.

Reply:
I mean a call like this: `Array.prototype.indexOf(some_array, NaN)`

Reply:
Yeah, i will add specific handlers for that since it is kind of misleading